tests/test_parse_requirements.py

```diff
@@ -66,12 +76,35 @@ def test_parse_requirements_with_index_url(monkeypatch, flag):
             "",
         ),
         (
+            # Version and URL can't be combined so this all gets parsed as a package name
             "pip==1.3.1@{url}\n".format(url=PEP508_PIP_EXAMPLE_URL),
             {"pip"},
             None,
             "pip==1.3.1@" + PEP508_PIP_EXAMPLE_URL,  # Note no extra space after @
             "==1.3.1@" + PEP508_PIP_EXAMPLE_URL,
         ),
+        (
+            PEP508_PIP_EXAMPLE_URL + "#egg=pip",
+            {"pip"},
+            None,
```

Comment:
Not sure what the expected behaviour is here. At the moment, we're just grabbing the package name from the `#egg=suffix` at the end. But it might be good to include the URL too.

If that's the case, we'd have do something like:
```python
# set the req to the egg fragment.  when it's not there, this
# will become an 'unnamed' requirement
req_str = _egg_fragment(link.url)
req_str = f"{req_str} @ {link.url}"
```